src/wp-includes/ms-functions.php

```diff
@@ -2321,6 +2321,18 @@ function add_new_user_to_blog(
 
 		if ( ! is_wp_error( $result ) ) {
 			update_user_meta( $user_id, 'primary_blog', $blog_id );
+
+			/**
+			 * Fires when a new user is added to a network site without an error from $result.
+			 *
+			 * @since 4.9.0
+			 *
+			 * @param int    $user_id  New user's ID.
+			 * @param mixed  $password New user's password.
+			 * @param array  $meta     Array of meta data from the new registration.
+			 * @param int    $blog_id  Blog ID.
+			 */
+			do_action( 'added_new_user_to_blog', $user_id, $password, $meta, $blog_id );
```

Comment:
Naming things: I'd go with the following as I think it a better practice to use the `wp_` prefix followed by the name of the function. I care more about the prefix that changing added to add.

```suggestion
			do_action( 'wp_add_new_user_to_blog', $user_id, $password, $meta, $blog_id );
```